serverless/lambda/src/main/java/uk/gov/di/helpers/CookieHelper.java

```diff
@@ -16,38 +15,36 @@
 
     public static Optional<SessionCookieIds> parseSessionCookie(Map<String, String> headers) {
         if (headers == null
-                || headers.isEmpty()
                 || !headers.containsKey(REQUEST_COOKIE_HEADER)
                 || headers.get(REQUEST_COOKIE_HEADER).isEmpty()) {
             return Optional.empty();
         }
-
-        final String COOKIE_REGEX = "gs=(?<sid>[^.;]+)\\.(?<csid>[^.;]+);";
-
-        String cookies = headers.getOrDefault(REQUEST_COOKIE_HEADER, "");
-
+        String cookies = headers.get(REQUEST_COOKIE_HEADER);
         LOGGER.debug("Session Cookie: {}", cookies);
+        HttpCookie httpCookie;
+        try {
+            httpCookie = HttpCookie.parse(cookies).stream().findFirst().get();
```

Comment:
Thinking about it, what are the chances of us getting multiple cookies on a single request? We might want a `filter(cookie -> "gs".equals(cookie.getName()))` or something

Reply:
Hmm good point